internal/providers/terraform/azure/postgresql_flexible_server.go

```diff
@@ -33,110 +30,39 @@ func NewAzureRMPostrgreSQLFlexibleServer(d *schema.ResourceData, u *schema.Usage
 	}
 
 	if len(s) > 2 {
-		tier = s[0]
+		tier = strings.ToLower(s[0])
 		size = s[2]
 	}
 
 	if len(s) > 3 {
 		version = s[3]
 	}
 
-	tierName := map[string]string{
-		"B":  "Burstable",
-		"GP": "General Purpose",
-		"MO": "Memory Optimized",
-	}[strings.ToUpper(tier)]
-
-	if tierName == "" {
-		log.Warnf("Unrecognised PostgreSQL tier prefix for resource %s: %s", d.Address, tierName)
```

Comment:
Decided to use the full `sku` value in the error messages as it reads better than separate substrings, the text already scopes them.